filterUsers.js

```diff
@@ -0,0 +1,124 @@
+                level: 1
+            }
+        ]
+    }
+];
+
+
+
+
+
+function nameORCompetence() {
+    console.log(document.getElementById("initialChoice").value)
+    if (document.getElementById("initialChoice").value == "name") {
+        document.getElementById("searchByName").removeAttribute("hidden", "");
+    } else if (document.getElementById("initialChoice").value == "competenze") {
+        document.getElementById("searchByCompetence").removeAttribute("hidden", "");
+
+    } else if (document.getElementById("initialChoice").value == "competenzeLivello") {
+        document.getElementById("searchByCompetenceAndLevel").removeAttribute("hidden", "");
+
+    }
+    document.getElementById("DIVinitialChoice").setAttribute("hidden", "")
+}
+
+
+let printSpace = document.getElementById("show");
+
+function searchByName() {
+    clear(printSpace);
+    for (let index = 0; index < users.length; index++) {
+        if (users[index].userName.toLowerCase().includes(document.getElementById("nomeCercato").value.toLowerCase())) {
+            document.getElementById("show").innerHTML += "<ul><br><li> Nome e Cognome: " + users[index].userName + "</li><br> <li> Competenze: <br> " + printArrayOfObject(users[index].competenceUser) + "</ul><br><br><hr>";
+        }
+    }
+}
+
+
+
+function searchByCompetence() {
+    clear(printSpace);
+    for (let index = 0; index < users.length; index++) {
+        for (let index2 = 0; index2 < users[index].competenceUser.length; index2++) {
+            let currentCompetence = users[index].competenceUser[index2].nameComp.toLowerCase();
+            let searchedCompetence = document.getElementById("competenzaCercata").value.toLowerCase();
+            if (currentCompetence == searchedCompetence) { //che mal di testa sa condizione
+                document.getElementById("show").innerHTML += "<ul><br><li> Nome e Cognome: " + users[index].userName + "</li><br> <li> Competenze: <br> " + printArrayOfObject(users[index].competenceUser) + "</ul><br><br><hr>";
+            }
+        }
+    }
+}
+
+
+
+function searchByCompetenceAndLevel() {
+    clear(printSpace);
+    for (let index = 0; index < users.length; index++) {
+        for (let index2 = 0; index2 < users[index].competenceUser.length; index2++) {
+            let currentCompetence = users[index].competenceUser[index2].nameComp.toLowerCase();
+            let searchedCompetence = document.getElementById("competenzaCercataLivello").value.toLowerCase();
+            let levelOfCompetence = users[index].competenceUser[index2].level;
+            let searchedLevelOfCompetence = parseInt(document.getElementById("levels").value);
+            if ((currentCompetence == searchedCompetence) && (levelOfCompetence >= searchedLevelOfCompetence)) { 
+                document.getElementById("show").innerHTML += "<ul><br><li> Nome e Cognome: " + users[index].userName + "</li><br> <li> Competenze: <br> " + printArrayOfObject(users[index].competenceUser) + "</ul><br><br><hr>";
```

Comment:
Sarebbe visivamente più comodo avere la stringa su più linee:

https://dev-tips.com/javascript/multi-line-strings-in-javascript